File: 프로그래머스/unrated/181839. 주사위 게임 1/주사위 게임 1.cs
using System;

public class Solution {
    public int solution(int a, int b) 
    {
        int oddCount = 0;
        if(a % 2 == 1) ++oddCount;
        if(b % 2 == 1) ++oddCount;
        
        if(oddCount == 2)
            return a * a + b * b;
        else if(oddCount == 1)
            return 2 * (a + b);
        else // oddCount == 0
            return Math.Abs(a - b);
    }
}

File: 프로그래머스/0/181839. 주사위 게임 1/주사위 게임 1.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

int solution(int a, int b) {
    
    int oddCount = (a & 1) + (b & 1); 
    
    if (oddCount == 2) return a * a + b * b;
    if (oddCount == 1) return 2 * ( a + b );
    else return abs(a - b);
}